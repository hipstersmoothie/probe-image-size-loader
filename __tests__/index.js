import fs from 'fs';
import probe from 'probe-image-size';

import probeImageSize from '../src';

test('probeImageSize - no loaders previous', () => {
  const buffer = Buffer.from(
    `module.exports = __dirname + 'some/path/to/image.png';`
  );

  jest.spyOn(fs, 'readFileSync').mockImplementationOnce(() => {});
  jest.spyOn(probe, 'sync').mockImplementationOnce(() => ({
    height: 100,
    width: 200
  }));

  expect(probeImageSize.bind({ resourcePath: '/' })(buffer)).toMatchSnapshot();
});

test('probeImageSize - some loaders previous', () => {
  const buffer = Buffer.from(
    `module.exports = {
      src: __dirname + 'some/path/to/image.png',
      lqip: 'path/to/lqip'
    };`
  );

  jest.spyOn(fs, 'readFileSync').mockImplementationOnce(() => {});
  jest.spyOn(probe, 'sync').mockImplementationOnce(() => ({
    height: 100,
    width: 200
  }));

  expect(probeImageSize.bind({ resourcePath: '/' })(buffer)).toMatchSnapshot();
});
