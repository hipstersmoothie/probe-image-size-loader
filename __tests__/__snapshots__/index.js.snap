// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`probeImageSize - no loaders previous 1`] = `
"
      module.exports = {
        src:  __dirname + 'some/path/to/image.png',
        height: 100,
        width: 200
      }
    "
`;

exports[`probeImageSize - some loaders previous 1`] = `
"module.exports = {
      src: __dirname + 'some/path/to/image.png',
      lqip: 'path/to/lqip'
   , height: 100, width: 200 };"
`;
